### DISCUSSION TITLE
Idea | Fusing on the GPU | octree/kdtree

### DISCUSSION THREAD

**Discussion by ghost** [cdcseacave/openMVS#1133] [Ideas]
Does this idea sound plausible? Creating a sparse/semi-dense version of the scene> bounding boxes from something like octree / kdtree> filtering or dividing views of each bounding box to not exceed GPU memory> fusing each scene using the GPU> combining all scenes using the CPU with an option for slicing the points to not exceed memory / point filtering.

**Comment by cdcseacave** [comment]
This is a valid idea (not sure why you closed it), and we could find a solution. However, taking a step back and looking at the high level picture, we might not even want the point cloud, we are interested in directly getting the mesh if possible. So how about developing a GPU fusion that takes as input a set of depth-maps and outputs a mesh, for ex something like TSDF with visibility information.

**Comment by ghost** [comment]
Sounds interesting, what would the process look like? Memory efficient cells enabling denser voxel grids, TSDFs for every depth map within a cell with interpolation on the GPU, then combining cells on the CPU/GPU?

**Reply by cdcseacave** [reply]
TSDF is powerful when fusing all depth-maps at once, so in order to fit the GPU memory I'd split the world space in smaller chunks and run TSDF on each of them, fusing all depth-maps overlapping with the chunk
